GLib2 tries to find "python3" and if it can't find it, it will go for "python"; if port select wasn't explicitly run, this will likely end-up with Python 2.7. As a fallback, meson can use whatever python it's running on if the argument to find_installation is empty.


--- meson.build.orig	2019-11-21 00:41:53.000000000 -0300
+++ meson.build	2019-11-21 00:44:00.000000000 -0300
@@ -1992,9 +1992,9 @@
 
 glib_conf.set('HAVE_PROC_SELF_CMDLINE', have_proc_self_cmdline)
 
-python = import('python').find_installation('python3')
+python = import('python').find_installation('')
 # used for '#!/usr/bin/env <name>'
-python_name = 'python3'
+python_name = '@PYTHON@'
 
 python_version = python.language_version()
 python_version_req = '>=3.4'
